### DISCUSSION TITLE
[Feature] Duplicate checking before uploading

### DISCUSSION THREAD

**Discussion by LucaTheHacker** [immich-app/immich#8728] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Currently, the duplicate check is performed server-side after the file is fully uploaded.

It would be great to have an HEAD request with the SHA256/512 of the file that we want to upload, so that the server could answer with 204 if the hash is known and with 404 is the hash is not known.

If the hash is known the file won't be uploaded, becoming particularly useful when a remote-backup is happening and the connection is not really fast.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
#4154
